src/goto-symex/builtin_functions.cpp

```diff
@@ -1704,3 +1618,26 @@ bool goto_symext::run_builtin(
 
   return false;
 }
+
+void goto_symext::replace_races_check(expr2tc &expr)
+{
+  if (!options.get_bool_option("data-races-check"))
+    return;
+
+  expr->Foreach_operand([this](expr2tc &e) {
+    if (!is_nil_expr(e))
+      replace_races_check(e);
+  });
+
+  if (is_races_check2t(expr))
+  {
+    // replace with __ESBMC_races_flag[address_of(var)]
+    const races_check2t &obj = to_races_check2t(expr);
+
+    expr2tc flag;
+    migrate_expr(symbol_expr(*ns.lookup("c:@F@__ESBMC_races_flag")), flag);
```

Comment:
Does it work for C++ programs, too?

Reply:
Yes, because this symbol is not created by the frontend.